open-access-variant-filtering-wf/main.nf

```diff
@@ -0,0 +1,199 @@
+]
+
+include { songScoreDownload as dnVcf } from './song-score-utils/song-score-download' params(download_params)
+include { metadataParser as mParser } from "./modules/raw.githubusercontent.com/icgc-argo/data-processing-utility-tools/metadata-parser.0.2.0.0/tools/metadata-parser/metadata-parser.nf"
+include { variantFilter as vFilter } from './wfpr_modules/github.com/icgc-argo/variant-calling-tools/variant-filter@0.1.0/main.nf' params(filter_params)
+include { payloadGenVariantProcessing as pGenVar } from "./wfpr_modules/github.com/icgc-argo/data-processing-utility-tools/payload-gen-variant-processing@0.1.0/main.nf" params(payloadGenVcf_params)
+include { songScoreUpload as upVcf } from './song-score-utils/song-score-upload' params(upload_params)
+include { getSecondaryFiles as getSec } from './wfpr_modules/github.com/icgc-argo/data-processing-utility-tools/helper-functions@1.0.0/main.nf'
+include { cleanupWorkdir as cleanup } from './wfpr_modules/github.com/icgc-argo/data-processing-utility-tools/cleanup-workdir@1.0.0/main.nf'
+
+
+// please update workflow code as needed
+workflow OpenFilterWf {
+  take:  // update as needed
+    study_id
+    analysis_id
+    vcf_file
+    analysis_metadata
+    regions_file
+
+
+  main:  // update as needed
+    local_mode = false
+    vcf = Channel.from()
+    vcf_idx = Channel.from()
+
+    if (analysis_id) {
+      // download vcf and metadata from song/score (analysis type: variant_calling)
+      dnVcf(study_id, analysis_id)
+      vcf = dnVcf.out.files.flatten().first()
+      vcf_idx = dnVcf.out.files.flatten().last()
+      vcf_meta = dnVcf.out.song_analysis
+    } else if (
+      !analysis_metadata.startsWith('NO_FILE') && \
+      !vcf_file.startsWith('NO_FILE')
+    ) {
+      local_mode = true
+      vcf = file(vcf_file)
+      vcf_idx = Channel.fromPath(getSec(vcf_file, ['tbi']))
+      vcf_meta = file(analysis_metadata)
+    } else {
+      exit 1, "To download input VCF files from SONG/SCORE, please provide `params.analysis_id`.\n" +
+                "Or please provide `params.analysis_metadata` and `params.vcf_file` to use local files as input."
+    }
+
+    // get analysis_tools info
+    mParser(vcf_meta)
+
+    mParser.out.analysis_tools.map{params.apply_filters[it]}.set{apply_filters}
+    mParser.out.analysis_tools.map{params.include[it]}.set{include}
+    mParser.out.analysis_tools.map{params.exclude[it]}.set{exclude}
+
+    // filter variants
+    vFilter(vcf, vcf_idx, file(regions_file), apply_filters, include, exclude, params.output_type)
+
+    // genPayloadSNV
+    pGenVar(vcf_meta, vFilter.out.filtered_vcf.concat(vFilter.out.filtered_vcf_tbi).collect(), name, short_name, version, params.open)
+
+    // skip upload if in local_mode
+    if (!local_mode) {
+        // uploadVariant
+        upVcf(study_id, pGenVar.out.payload, pGenVar.out.files_to_upload)
+    }
+
+    // cleanup, skip cleanup when running in local mode
+    if (params.cleanup && !local_mode) {
```

Comment:
I know we were doing this (ie, skipping cleanup for local mode) before. Now since we have the publish_dir param, the needed output files will be copied to  the final output specified by workflow's publish_dir, we should only let `params.cleanup` to control whether to perform clean up or not regardless which mode.